### PR TITLE
Broken Regressions Fixes

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Fix the currently broken regressions tests

#### Related Issue

N/A

#### Motivation and Context

Some of our regression tests that we rely on for testing Lean are broken, thankfully there are only a few and seem to be related to recent additions such as #5389, #5381, and #5379 

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

After these updates all regression pass as expected except two.

These two are `BasicTemplateIndexAlgorithm` and `IndexOptionCallITMGreeksExpiryRegressionAlgorithm`

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by C-SELLERS** [comment]
Hey @gsalaz98 , do you mind checking out the two remaining broken regressions? 

- `BasicTemplateIndexAlgorithm`
-  `IndexOptionCallITMGreeksExpiryRegressionAlgorithm`


I think they were broken on addition and I'm not sure how they should be acting 😄

**Comment by gsalaz98** [comment]
> 
> 
> Hey @gsalaz98 , do you mind checking out the two remaining broken regressions?
> 
>     * `BasicTemplateIndexAlgorithm`
> 
>     * `IndexOptionCallITMGreeksExpiryRegressionAlgorithm`
> 
> 
> They were broken on addition and I'm not sure how they should be acting 😄

`IndexOptionCallITMGreeksExpiryRegressionAlgorithm` is expected to be broken for now, but I think I never got around to updating `BasicTemplateIndexAlgorithm` w/ the push for index options.

File: Algorithm.CSharp/IndexOptionCallOTMExpiryRegressionAlgorithm.cs [hunk 1]
```diff
@@ -165,7 +165,7 @@ public override void OnEndOfAlgorithm()
         /// <summary>
         /// This is used by the regression test system to indicate which languages this algorithm is written in.
         /// </summary>
-        public Language[] Languages { get; } = { Language.CSharp, Language.Python };
+        public Language[] Languages { get; } = { Language.CSharp };
```
Comment (jaredbroad) [inline Algorithm.CSharp/IndexOptionCallOTMExpiryRegressionAlgorithm.cs:168]: @gsalaz98 please add a commit to make sure these pass in python as well

**Review by jaredbroad — CHANGES_REQUESTED** [review]
Tests should run in python

**Review by jaredbroad — APPROVED** [review]

**Review by jaredbroad — APPROVED** [review]

**Pull request merged**